src/wp-includes/robots-template.php

```diff
@@ -0,0 +1,171 @@
+<?php
+/**
+ * Robots template functions.
+ *
+ * @package WordPress
+ * @subpackage Robots
+ * @since 5.6.0
+ */
+
+/**
+ * Displays the robots meta tag as necessary.
+ *
+ * Gathers robots directives to include for the current context, using the
+ * {@see 'wp_robots'} filter. The directives are then sanitized, and the
+ * robots meta tag is output if there is at least one relevant directive.
+ *
+ * @since 5.6.0
+ */
+function wp_robots() {
+	/**
+	 * Filters the directives to be included in the 'robots' meta tag.
+	 *
+	 * The meta tag will only be included as necessary.
+	 *
+	 * @since 5.6.0
+	 *
+	 * @param array $robots Associative array of directives. Every key must be the name of the directive, and the
+	 *                      corresponding value must either be a string to provide as value for the directive or a
+	 *                      boolean `true` if it is a boolean directive, i.e. without a value.
+	 */
+	$robots = apply_filters( 'wp_robots', array() );
+
+	// TODO: Implement additional sanitization.
```

Comment:
What sanitization are you thinking about here? Should we include an allowed-list and only accept specific directives?

Reply:
Hmm I'm not entirely sure, but we could ensure that some other opposing directives cannot be set at the same time (like `index` and `noindex` etc). Alternatively we could start without sanitization entirely and think about it later.